src/Hosting/Hosting/src/Internal/HostingRequestStartingLog.cs

```diff
@@ -58,18 +45,8 @@ public override string ToString()
         {
             if (_cachedToString == null)
             {
-                _cachedToString = string.Format(
-                    CultureInfo.InvariantCulture,
-                    "Request starting {0} {1} {2}://{3}{4}{5}{6} {7} {8}",
-                    _request.Protocol,
-                    _request.Method,
-                    _request.Scheme,
-                    _request.Host.Value,
-                    _request.PathBase.Value,
-                    _request.Path.Value,
-                    _request.QueryString.Value,
-                    _request.ContentType,
-                    _request.ContentLength);
+                var request = _request;
+                _cachedToString = $"{LogPreamble}{request.Protocol} {request.Method} {request.Scheme}://{request.Host.Value}{request.PathBase.Value}{request.Path.Value}{request.QueryString.Value} {EscapedValueOrEmptyMarker(request.ContentType)} {ValueOrEmptyMarker(request.ContentLength)}"; ;
```

Comment:
@benaadams I just tried this locally before merging, and I noticed that the content-type in the request finished logged is getting URL-encoded. Should we really be escaping the content-type? Notice that `text/plain` becomes `text%2Fplain` below.

#### Current Master Log

```
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 76.3405ms 200 text/plain
```

#### Betterized Hosting Log

```
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished HTTP/1.1 GET http://localhost:5000/ - - - 200 14 text%2Fplain 50.6745ms
```

Reply:
Should be EscapeUriString

Reply:
Hmm; looking at IIS logs it replaces space with `+` but EscapeUriString does `%20`

Reply:
Spaces are significant in the log; so `text/plain; charset=UTF-8` needs it space escaping `text/plain;+charset=UTF-8` checking what encoding does that.

Reply:
Looks like I need `WebUtility.UrlEncode`

Reply:
Oh, that makes it terrible in .NET Core
`text%2Fplain%3B+charset%3DUTF-8`

Reply:
@GrabYourPitchforks is there a better way of doing this? https://github.com/aspnet/AspNetCore/pull/10102/commits/2bdc4e706b06055e61f4d3d70332d9429965cbec

Reply:
Checking IIS logs it just converts space to `+` but leaves `+` as `+`

Reply:
@halter73 should be good now